sematic/retry_settings.py

```diff
@@ -0,0 +1,41 @@
+# Standard Library
+from dataclasses import dataclass, field
+from typing import Tuple, Type
+
+# Sematic
+from sematic.utils.exceptions import ExceptionMetadata
+
+
+@dataclass
+class RetrySettings:
+    """
+    Configuration object to pass to `@sematic.func` to activate retries.
+
+    Parameters
+    ----------
+    exceptions: Tuple[Type[Exception]]
+        A tuple of exception types to retry for.
+
+    times: int
+        How may times to retry.
+    """
+
+    exceptions: Tuple[Type[Exception]]
+    times: int
+    retry_count: int = field(default=0, init=False)
+
+    def should_retry(self, exception_metadata: ExceptionMetadata) -> bool:
+        """
+        Should the given exception trigger a retry?
+        """
+        if not self._matches_exceptions(exception_metadata):
+            return False
+
+        return self.retry_count < self.times
```

Comment:
For "2 times means you do it once, then again if it fails, and no more" this should be `self.retry_count <= self.times`

Reply:
Yes Tudor is correct. It's the number of retries, not the number of tries.